server/server.go

```diff
@@ -573,10 +573,17 @@ func (s *Server) startReloadWatcher(ctx context.Context) {
 		for {
 			// Block until the signal is received
 			<-ch
-			if err := s.config.Reload(); err != nil {
+			newConfig, err := s.config.Reload()
+			if err != nil {
 				logrus.Errorf("Unable to reload configuration: %v", err)
 				continue
 			}
+			if s.config.IsPinnedImagesUpdated(newConfig) {
```

Comment:
It's not clear to me how this condition will be true if we're replacing the config in place in Reload(). won't newConfig == s.config at this point?

Reply:
The `Reload()` does not replace the config; it only updates the `newConfig` variable, and all the changes vanish with returns.

https://github.com/cri-o/cri-o/blob/0ae4367ef3b337894b13cd199b007653a5c2b6be/pkg/config/reload.go#L26-L38

Reply:
wait but we do replace the c.config in reload: https://github.com/cri-o/cri-o/blob/0ae4367ef3b337894b13cd199b007653a5c2b6be/pkg/config/reload.go#L148-L156 we absorb the newConfig, and then take select changes into the c.config (old config) so how would the newConfig and c.config ever differ  WRT pinned images?

Reply:
We're merging the old config pinned image list with the new pinned image list, and if there are changes, we update the pinned image list in `imageService`. Because we're relying on the returned state from `ReloadPinnedImages` we can perform `imageService` update only outside the Reload method.